Initial commit of collective_utils

amrshennawi opened the pull request:
Summary:
Details in T133020932
First commit of collective utils library. Ported over from model store, removed scuba logging, error_trait and all dependencies on modelstore.

Test Plan: In the following diffs.

Differential Revision: D45545970

H-Huang reviewed (approved):
Reviewed and stamped internally

amrshennawi committed:
Initial commit of collective_utils (#101037)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/101037

Details in T133020932
First commit of collective utils library. Ported over from model store, removed scuba logging, error_trait and all dependencies on modelstore.

Test Plan: buck2 test mode/dev-nosan caffe2/test/distributed:collective_utils

Reviewed By: H-Huang

Differential Revision: D45545970

fbshipit-source-id: fed43bcb7cc03a6ddf687bd24ea7c3708369a4bd
